Stevenxl implement group assignment

StevenXL opened the pull request.

StevenXL committed:
Add route to accept a user's address input.

StevenXL committed:
Add logic to Groups controller to add a user to a specific group.

StevenXL committed:
Save user after assigning to a group.

StevenXL committed:
Modify Group model so that it has the attributes we expect a user to
sign up with.

StevenXL committed:
Change groups migration so that the attributes we expect are saved.

Note that the name is no longer a field on our group table, but it is a
virtual attribute in our model. We no longer have uniqueness validations
at the Model level or constraints at the Database level.

kimchanyoung commented on app/controllers/groups_controller.rb:
```diff
@@ -0,0 +1,15 @@
+  def assign
+    user = current_user
+    user.group_id = find_group(params[:address])
+    user.save
```
Do we want to write a case for when (for whatever reason) the group doesn't save?

kimchanyoung commented on app/models/group.rb:
```diff
@@ -1,5 +1,11 @@
 
-  validates :name, :address, presence: true, uniqueness: true
+  validates :street, :city, presence: true
+  validates :number, :zip_code,  numericality: { only_integer: true }
```
Are there any benefits of having the zipcode be an integer as opposed to a string?

kimchanyoung commented on app/controllers/groups_controller.rb:
```diff
@@ -0,0 +1,15 @@
+  private
+
+  def find_group(address_hash)
+    Group.where(address_hash).pluck(id)
```
just out of curiosity, what are the benefits of using where->pluck vs find(hash[id])

StevenXL committed:
Add logic to redirect user if the save fails.

StevenXL committed:
Accept number and zip_code as strings, not integer.

StevenXL committed:
Convert type of number and zip_code fields to string from integer.

StevenXL committed:
Removed indeces on address column of the group table. This column no
longer exists.

We can add a uniqueness constraint on multiple columns if we wish to
have a uniqueness constraint.

StevenXL committed:
Update schema.rb.

StevenXL committed:
Use the find_by method in the find_group method instead of where.

where returns a collection of objects, even if that collection is only
one item. find_by returns only one object.
